tests/fuzz/s2n_hybrid_ecdhe_kyber_r3_fuzz_test.c

```diff
@@ -55,27 +57,47 @@ static int setup_connection(struct s2n_connection *server_conn)
     server_conn->secure->cipher_suite = &s2n_ecdhe_kyber_rsa_with_aes_256_gcm_sha384;
     server_conn->handshake_params.conn_sig_scheme = s2n_rsa_pkcs1_sha384;
 
-    POSIX_GUARD(s2n_dup(&server_kem_params.private_key, &server_conn->kex_params.kem_params.private_key));
+    POSIX_GUARD(s2n_dup(&params->private_key, &server_conn->kex_params.kem_params.private_key));
     POSIX_GUARD(s2n_ecc_evp_generate_ephemeral_key(&server_conn->kex_params.server_ecc_evp_params));
 
     return S2N_SUCCESS;
 }
 
+int s2n_fuzz_init_kem_param(struct s2n_kem_params *param) {
+    POSIX_ENSURE_REF(param->kem);
+    struct s2n_blob *public_key_1 = &param->public_key;
+    POSIX_GUARD(s2n_alloc(public_key_1, S2N_KYBER_512_R3_PUBLIC_KEY_BYTES));
+
+    /* Do not GUARD this call to s2n_kem_generate_keypair(), as it will fail when attempting to generate the KEM
+     * key pair if !s2n_pq_is_enabled(). However, even if it does fail, it will have allocated zeroed memory for the
+     * public and private keys needed for setup_connection() to complete. */
+    s2n_result result = s2n_kem_generate_keypair(param);
```

Comment:
Why wasn't this true before? Is this a behavior change?

Reply:
+1
If PQ is disabled, why run this test at all?

Reply:
These code changes are keeping the previously existing behavior of the fuzz test, and updating the initialization function to successfully initialize both Kyber parameters instead of only the first one.

Previously when running this fuzz test with `Openssl-1.0.2-fips` (where PQ is automatically disabled), `s2n_kem_generate_keypair()` would allocate zeroed memory for the Kyber key pairs but [eventually fail with `S2N_ERR_PQ_DISABLED`](https://github.com/aws/s2n-tls/blob/5d1898a93809c0fe05233059400fcf5987171ef9/pq-crypto/kyber_r3/kyber512r3_kem.c#L29) and return `S2N_FAILURE` from `s2n_fuzz_init()`. This non-zero return code from the fuzz init function does NOT cause libfuzzer to skip running `s2n_fuzz_test()`, and libfuzzer will still fuzz the `s2n_client_key_recv()` function. `s2n_client_key_recv()` will call into both the ECDHE and PQ code, but with a zeroed Kyber key-pair. Since `s2n_pq_is_enabled()` will return false in this mode, if `s2n_client_key_recv()` gets far enough to call into Kyber's decapsulate method, it will also [immediately fail with `S2N_ERR_PQ_DISABLED`](https://github.com/aws/s2n-tls/blob/5d1898a93809c0fe05233059400fcf5987171ef9/pq-crypto/kyber_r3/kyber512r3_kem.c#L114), and the zeroed Kyber key-pair will never actually be used by `s2n_client_key_recv()`.

So when PQ is disabled this fuzz test is still fuzzing the ECDHE half of the hybrid key share. If the ECDHE half is successful, the PQ portion is guaranteed to eventually fail with `S2N_ERR_PQ_DISABLED`. 

If we attempt to skip this test altogether by adding `if (!s2n_pq_is_enabled()) { return S2N_SUCCESS; }` to the first line of `s2n_fuzz_test()`, it will trigger [our fuzz test safety checks](https://github.com/aws/s2n-tls/blob/2352dd1039b0023716f31a6a05ae1a105a252017/tests/fuzz/runFuzzTest.sh#L200-L202) that ensures each fuzz test covers a minimum number of branches, and will actually cause the test to start failing instead of being skipped. Ideally, long term, we should have a better way to skip running fuzz tests based on runtime checks, but for now, keeping this fuzz tests behavior as-is still provides some value due to it still fuzzing the ECDHE half when PQ is disabled, and fuzzing both halves in all other configurations.